tests/sentry/middleware/test_devtoolbar.py

```diff
@@ -0,0 +1,219 @@
+        self.middleware(request)
+
+        mock_record.assert_called()
+        assert mock_record.call_args[0][0] == self.analytics_event_name
+        assert mock_record.call_args[1].get("origin") == origin
+
+    @patch("sentry.analytics.record")
+    def test_response_status_code(self, mock_record: MagicMock):
+        request = self.make_toolbar_request()
+        self.get_response.return_value = HttpResponse(status=420)
+        self.middleware(request)
+
+        mock_record.assert_called()
+        assert mock_record.call_args[0][0] == self.analytics_event_name
+        assert mock_record.call_args[1].get("status_code") == 420
+
+    @patch("sentry.analytics.record")
+    def test_methods(self, mock_record: MagicMock):
+        for method in ["GET", "POST", "PUT", "DELETE"]:
+            request = self.make_toolbar_request(method=method)
+            self.middleware(request)
+
+            mock_record.assert_called()
+            assert mock_record.call_args[0][0] == self.analytics_event_name
+            assert mock_record.call_args[1].get("method") == method
+
+
+TEST_MIDDLEWARE = (
+    "django.middleware.common.CommonMiddleware",
+    "django.contrib.sessions.middleware.SessionMiddleware",
+    "sentry.middleware.auth.AuthenticationMiddleware",
+    "sentry.middleware.devtoolbar.DevToolbarAnalyticsMiddleware",
+)
+
+
+class DevToolbarAnalyticsMiddlewareIntegrationTest(APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.login_as(user=self.user)
+        self.origin = "https://third-party.site.com"
+
+    @override_settings(MIDDLEWARE=TEST_MIDDLEWARE)
+    @patch("sentry.analytics.record")
+    def _test_e2e(
+        self,
+        view_name: str,
+        method: str,
+        url_params: dict[str, str],
+        query_string: str,
+        mock_record: MagicMock,
+    ):
+        url = reverse(view_name, kwargs=url_params)
+        url += query_string
+        response: HttpResponse = getattr(self.client, method.lower())(
+            url,
+            headers={
+                "queryReferrer": "devtoolbar",
+                "Origin": self.origin,
+            },
+        )
+
+        org_id, org_slug = parse_id_or_slug_param(url_params.get("organization_id_or_slug"))
+        proj_id, proj_slug = parse_id_or_slug_param(url_params.get("project_id_or_slug"))
+
+        mock_record.assert_called_with(
+            "devtoolbar.api_request",
+            view_name=view_name,
+            route="^api/0/organizations/(?P<organization_id_or_slug>[^\\/]+)/replays/$",
+            query_string=query_string,
+            origin=self.origin,
+            method=method,
+            status_code=response.status_code,
+            organization_id=org_id,
+            organization_slug=org_slug,
+            project_id=proj_id,
+            project_slug=proj_slug,
+            user_id=self.user.id,
+        )
+
+    def test_e2e_replays(self):
```

Comment:
Planning to test with some other endpoints, ex issue details and project user feedback. Just want to confirm this is the right approach first